### PR TITLE
Add docstrings and comments

### LINKED ISSUE #1: Add code documentation to mathemagic.py
Add comments and docstrings to the mathemagic.py to explain to a user what the script is doing and how it works

### PR DESCRIPTION
**Pull request by eddyannobil**
This commit adds comments and docstrings that further explains what the code does and how it works.
Resolves: #1

### CODE REVIEW

File: mathemagic.py [hunk 1]
```diff
@@ -1,8 +1,15 @@
 #!/usr/bin/env python3
+'''
+This script computes the actual sum of the five "random" four-digit numbers 
+to cross-check with the predicted sum and display the result to the user.
+'''
+
+#A variable is declared to store the user's name and input number
```
Comment (fiifikrampah) [inline mathemagic.py]: All comments should have a space between the comment sign and the actual comment.

File: mathemagic.py [hunk 2]
```diff
@@ -29,9 +36,12 @@
             random_num = ('20' + str(random_number))
         else:
             random_num = ('2' + str(random_number)) 
+#The sum of all randoms numbers is predicted at the start of the first input
```
Comment (fiifikrampah) [inline mathemagic.py]: This comment should be indented. Also, there is a typo in there. The comment should read: 

> The sum of all random numbers is predicted at the start of the first input

File: mathemagic.py [hunk 3]
```diff
@@ -29,9 +36,12 @@
             random_num = ('20' + str(random_number))
         else:
             random_num = ('2' + str(random_number)) 
+#The sum of all randoms numbers is predicted at the start of the first input            
         print(f"The sum of all the numbers you will enter will be {random_num}")    
-        random_no2 = input("Enter another 4 digit number:")
 
+        
+        random_no2 = input("Enter another 4 digit number:")
+#A second random number is received if it is valid
```
Comment (fiifikrampah) [inline mathemagic.py]: This comment should be indented too

File: mathemagic.py [hunk 4]
```diff
@@ -62,6 +72,7 @@
             print(f"{random_no2} is not a number") 
 
         random_no3 = (input("Enter another 4 digit number:"))
+#A third random number is received if it is valid
```
Comment (fiifikrampah) [inline mathemagic.py]: Same as this one

File: mathemagic.py [hunk 5]
```diff
@@ -91,6 +102,7 @@
                     print(f"So far, the numbers we have are: {random_no}, {random_no2}, {nine_minus_random_a}, {random_no3}, {nine_minus_random_b}")
                     print("Take a sec to compute the sum of these numbers...")
                     sum_of_4 = int(random_no) + int(random_no2) + int(random_no3) + int(nine_minus_random_a) + int(nine_minus_random_b)
+#The sum of all 5 random numbers is predicted
```
Comment (fiifikrampah) [inline mathemagic.py]: And this too

File: mathemagic.py [hunk 6]
```diff
@@ -29,9 +36,12 @@
             random_num = ('20' + str(random_number))
         else:
             random_num = ('2' + str(random_number)) 
+#The sum of all randoms numbers is predicted at the start of the first input            
         print(f"The sum of all the numbers you will enter will be {random_num}")    
-        random_no2 = input("Enter another 4 digit number:")
 
+
```
Comment (fiifikrampah) [inline mathemagic.py]: Nit: This space seems unneeded

**Review by fiifikrampah — CHANGES_REQUESTED** [review]
@eddyannobil Please address these slight issues and we should be good to merge. Thanks!

**Comment by eddyannobil** [comment]
@fiifikrampah I updated the PR and addressed all the review made

File: mathemagic.py [hunk 7]
```diff
@@ -29,9 +36,11 @@
             random_num = ('20' + str(random_number))
         else:
             random_num = ('2' + str(random_number)) 
+        # The sum of all random numbers is predicted at the start of the first input            
         print(f"The sum of all the numbers you will enter will be {random_num}")    
-        random_no2 = input("Enter another 4 digit number:")
```
Comment (fiifikrampah) [inline mathemagic.py:33]: Hmm let's avoid changing this line as it's not quite a part of documentation.

**Review by fiifikrampah — CHANGES_REQUESTED** [review]

**Comment by eddyannobil** [comment]
The unnecessary space on that line was removed

**Review by fiifikrampah — APPROVED** [review]
@eddyannobil Looks good! We can merge this now.

**Pull request merged**